rootfs/var/lib/dietpi/services/fs_partition_resize.sh

```diff
@@ -39,6 +39,26 @@
 		exit 1
 	fi
 
+	# check if the last partition is a 4MB partition with a vfat filesystem
+	LAST_FS_TYPE=$(lsblk -no FSTYPE "$ROOT_DRIVE" | tail -1)
+	if sfdisk -l "$ROOT_DRIVE" | tail -1 | grep -E "\s4M\s" > /dev/null 2>&1 && [[ $LAST_FS_TYPE = 'vfat' ]]
+	then
+		# the last partition is a 4M FAT filesystem - let's check if it is ours
+		SETUP_PART=$(sfdisk -l "$ROOT_DRIVE" | tail -1 | mawk '{print $1}')
+		if blkid "$SETUP_PART" | grep 'LABEL="DIETPISETUP"'
```

Comment:
Shall we make it a little simpler and just check for the label, using one which is as unique as possible? "DIETPISETUP" is probably sufficiently unique, otherwise "TMPDIETPITXTPART" or something?

Generally this can be used to do all checks in a single line:
```sh
[[ $(lsblk -nrbo FSTYPE,SIZE,LABEL /dev/mmcblk0 | tail -1) == 'vfat 4194304 DIETPISETUP' ]]
```
But maybe we should/can skip the size when using a sufficiently unique label, so we can change it later in a single place when needed. E.g. currently 1 MiB would be sufficient, but we might want to add the possibility to add further configs files or flags there at a later time. I have some in mind we should handle already now, if present: `dietpiEnv.txt`, `config.txt`, `unattended_pivpn.conf`, `Automation_Custom_PreScript.sh`, `Automation_Custom_Script.sh`.

And probably we should add a readme which briefly explains the partition? A little note that one must take care to preserve/use UNIX/LF line endings in text files might be helpful as well, or we convert them on copy?

Reply:
> Shall we make it a little simpler and just check for the label, using one which is as unique as possible? "DIETPISETUP" is probably sufficiently unique, otherwise "TMPDIETPITXTPART" or something?

You have only 11 characters.
That's why I picked this. It's 11 characters and tells you what it's for.

> Generally this can be used to do all checks in a single line:
> ```sh
> [[ $(lsblk -nrbo FSTYPE,SIZE,LABEL /dev/mmcblk0 | tail -1) == 'vfat 4194304 DIETPISETUP' ]]
> ```

Neat. I like it much better than my attempt.

> But maybe we should/can skip the size when using a sufficiently unique label, so we can change it later in a single place when needed. E.g. currently 1 MiB would be sufficient, but we might want to add the possibility to add further configs files or flags there at a later time. I have some in mind we should handle already now, if present: `dietpiEnv.txt`, `config.txt`, `unattended_pivpn.conf`, `Automation_Custom_PreScript.sh`, `Automation_Custom_Script.sh`.
> 
> And probably we should add a readme which briefly explains the partition? A little note that one must take care to preserve/use UNIX/LF line endings in text files might be helpful as well, or we convert them on copy?

All excellent points. I picked 4MB almost randomly. You are right that 1MB could hold most things that we'd ever consider.
And adding the Readme also makes perfect sense.

Ironically, I don't have easy access to a Windows system, so I don't know if it would mangle like endings. It might be better to "fix it in post", i.e. make sure this is correct when copying.

I'm paranoid, so I'd still want to have a fixed list of files that are supported. Is the list above what you'd suggest?

Reply:
On common editors on a non-ancient Windows, LF line endings are preserved. We had an automatic fix as failsafe step in the past for the boot FAT partition (where present) but I removed it a long time ago and remember only a single case in years where one used an unusual editor. It didn't seem worth doing file manipulations. However, here we do file copy/writes anyway, this feature explicitly addresses Windows (and macOS) users, and as of the other optional files one might create new ones, where Windows editors pick CRLF endings by default.

I fully agree with a fixed list of supported files, so jep, the above list should be complete for now, and those are present either depending on the hardware (like `config.txt` only on RPi, ... wait, there we do not need that FAT, remove it from the list 😅) or optional for additional first boot install automation.